feature-currency/src/test/java/com/stefang/app/feature/currency/viewmodel/CurrencyConverterViewModelTest.kt

```diff
@@ -150,4 +149,33 @@ class CurrencyConverterViewModelTest {
 
         collectJob.cancel()
     }
+
+    @Test
+    fun whenTrackHistory_thenHistoryRepositoryHasLatestRecord() = runTest {
+        historyRepository.histories.test {
+            setUp()
+            viewModel.trackHistory("USD", 10000)
+            testAppDispatchers.advanceUntilIdle()
+            val history = awaitItem().first()
+            assertEquals("USD", history.code)
+            assertEquals(10000, history.amount)
+
+            cancelAndConsumeRemainingEvents()
+        }
+    }
+
+    @Test
+    fun whenTrackHistoryHappenTwice_thenHistoryRepositoryHasLatestRecord() = runTest {
+        historyRepository.histories.test {
+            setUp()
+            viewModel.trackHistory("USD", 10000)
+            viewModel.trackHistory("IDR", 20000)
+            testAppDispatchers.advanceUntilIdle()
```

Comment:
aku jg punya method `advanceTimeBy` di atas, but somehow gk jalan. pke `advanceUntilIdle` baru bisa

Reply:
Hmm AFAIK kalo kita ga urusan dengan delay or specific time related coroutine, kita ga harus pake interasi dengan scehduler langsung.

Reply:
Anyway, it's kinda confusing to have so many dispatchers around:

1. `CoroutineDispatcher` from  `TestAppDispatchers`
2. `CoroutineDispatcher` from `dispatcherRule` 

Can we simplify this?